lib/fs.js

```diff
@@ -101,6 +101,10 @@ function nullCheck(path, callback) {
   return true;
 }
 
+function isFd(path) {
+  return (path >>> 0) === path && path >= 0;
```

Comment:
I generally avoid coercion as much as possible. It usually leads to rather unclear behavior. Throwing on `NaN` sounds like it would be expected. There is no sensible default we can choose in that case.